### PR TITLE
fix(api): get projects from provider with username

### PR DESCRIPTION
**Pull request by fsamin**
1. Description
1. Related issues
1. About tests
1. Mentions

@ovh/cds

### CODE REVIEW

File: engine/api/project.go [hunk 1]
```diff
@@ -83,74 +163,22 @@ func (api *API) getProjectsHandler() service.Handler {
...
+		var groupIDs []int64
+		if requestedUser == nil {
+			groupIDs = getAPIConsumer(ctx).GetGroupIDs()
+		} else {
+			groupIDs = requestedUser.GetGroupIDs()
+		}
+
 		pKeys := projects.Keys()
-		perms, err := permission.LoadProjectMaxLevelPermission(ctx, api.mustDB(), pKeys, getAPIConsumer(ctx).GetGroupIDs())
+		perms, err := permission.LoadProjectMaxLevelPermission(ctx, api.mustDB(), pKeys, groupIDs)
```
Comment (richardlt) [inline engine/api/project.go]: Looks like this code is useful only if withPermission, also maintainer an admin can have no permission on project based on groups but we should add read permission for maintainer and all permissions for admin.

File: engine/api/project.go [hunk 2]
```diff
@@ -27,12 +27,93 @@ import (
...
+		}
+	} else {
+		projects, err = project.LoadAllByRepoAndGroupIDs(ctx, api.mustDB(), api.Cache, getAPIConsumer(ctx).GetGroupIDs(), filterByRepo, opts...)
+		if err != nil {
+			return err
+		}
+	}
+
+	pKeys := projects.Keys()
+	perms, err := permission.LoadProjectMaxLevelPermission(ctx, api.mustDB(), pKeys, getAPIConsumer(ctx).GetGroupIDs())
```
Comment (richardlt) [inline engine/api/project.go]: Same comments

File: engine/api/project.go (hunk 1 above)
Comment (fsamin) [reply engine/api/project.go]: We always need permission. BWT i fixed it for admin, and maintainer permissions

File: engine/api/project.go (hunk 2 above)
Comment (fsamin) [reply engine/api/project.go]: fixed

**Pull request merged**